sensu/plugins/check-network-traffic.rb

```diff
@@ -0,0 +1,65 @@
+#!/opt/sensu/embedded/bin/ruby
+#
+#
+
+require 'rubygems' if RUBY_VERSION < '1.9.0'
+require 'sensu-plugin/check/cli'
+
+class CheckNetworkStats < Sensu::Plugin::Check::CLI
+
+  linkspeed = `ethtool eth0 | grep Speed`
+  puts linkspeed
+
+  if linkspeed.include? "10000Mb"
+  linkwarn = 9000
+  linkcrit = 12000
+  else
+  linkwarn = 900
+  linkcrit = 1200
+  end
+
+  option :warn,
+    :short => '-w WARN',
+    :proc => proc {|a| a.to_i },
+    :default => linkwarn
+  option :crit,
+    :short => '-c CRIT',
+    :proc => proc {|a| a.to_i },
+    :default => linkcrit
+  option :rxmcswarn,
+    :short => '-pw WARN',
+    :proc => proc {|a| a.to_i },
+    :default => 9000
+  option :rxmcscrit,
+    :short => '-pc CRIT',
+    :proc => proc {|a| a.to_i },
+    :default => 10000
+  option :iface,
+    :short => '-i IFACE',
+    :default => "eth0"
+
+def run
+     iface = config[:iface]
+    `sar -n DEV 5 1 | grep Average | grep "#{iface} "`.each_line do |line|
```

Comment:
Whitespace isn't aligned. I would prefer a single grep or filter in Ruby, but this works.

Hint:
```
`sar -n DEV 5 1`.lines.select{|k| k =~ /^Average:\s+#{ifname}\b/ }
```